src/Plutus/Certification/Local.hs

```diff
@@ -67,6 +67,7 @@ localServerCaps backend = do
 
     getRuns _ (RunID jobId) =
       Map.findWithDefault [] jobId <$> (lift $ readIORef jobs) >>= yieldMany
+    abortRuns = error "not implemented"
```

Comment:
Instead of the line marked "Purposefully leak" above, we can store the return from `async` in the state and then kill the async run from here. Processes should be killed, tempdirs cleaned up, etc.

Reply:
done.

NOTE: I added another `IORef` for cancellations. The other two options would be to use the same `IORef`, but  IMO the code looks much more straightforward.  Do you think we should clean the Map entry after the job is finished?

Reply:
Can't clean the status map entry since you need to check the status after the job is done, but yeah should be safe to remove the `Async` afterward.

Reply:
I added the modifications for cleaning the cancellation-map